main.py

```diff
@@ -111,9 +112,122 @@ def compute_mutual_information(training_set, event1, event2):
+
+        # v has been checked
+        vertexes_to_check_edge_orientation.remove(v)
+        checked_vertexes += v
+
+    # The graph is now directed
+    result._directed = True
+
+    return result
+
+
+def create_directed_fault_tree(undirected_fault_tree: Graph, top_event: Vertex):
+    """
+    Creates a directed fault tree out of an undirected fault tree.
+    Doesn't use a deepcopy yet, which may give some problems
+
+    :return: A directed Fault Tree of a bow tie.
+    """
+    result = undirected_fault_tree
+
+    orient_top_event_for_fault_tree(top_event)
+
+    return result
+
+
+def is_leaf(vertex: Vertex):
+    return len(vertex.incidence) > 1
+
+
+def orient_top_event_for_fault_tree(parent: Vertex):
+    for e in parent.incidence:
+        check_edge_direction_event_tree(e, e.other_end(parent))
+        if not is_leaf(e.other_end(parent)):
+            check_potential_articulation_point_fault_tree(e)
+
+
+def treat_as_parent(child: Vertex, articulation_point: Vertex):
+    for child_edge in child.incidence:
+        if not child_edge.other_end() == articulation_point:
+            check_edge_direction_event_tree(child_edge, child_edge.other_end(child))
+            if not is_leaf(child_edge.other_end(child)):
+                check_potential_articulation_point_fault_tree(child_edge)
+
+
+def determine_independence_of_parent_and_child(parent: Vertex, articulation_point: Vertex, child: Vertex):
+    yes_or_no = input("Are " + parent.label + " and " + child.label + "independend? (y/n)")
+    while not (yes_or_no == "y" or yes_or_no == "n"):
+        yes_or_no = input("Wrong input! Are " + parent.label + " and " + child.label + "independend? (y/n)")
+    if yes_or_no == "y":
+        check_edge_direction_event_tree(e, articulation_point)
+        if not is_leaf(child):
+            treat_as_parent(child, articulation_point)
+    elif yes_or_no == "n":
+        check_edge_direction_event_tree(e, child)
+        if not is_leaf(child):
+            check_potential_articulation_point_fault_tree(e)
+
+
+def check_potential_articulation_point_fault_tree(parent_edge: Edge):
+    parent = parent_edge.head
+    articulation_point = parent_edge.tail
+    for e in articulation_point.incidence:
+        child = e.other_end(articulation_point)
+        if not child == parent:
+            determine_independence_of_parent_and_child(parent, articulation_point, child)
+
+
+def get_potential_articulation_point_from_just_oriented_edge(edge: Edge, parent: Vertex):
+    """
+    Returns a list of edge tuples (parent_edge and child_edge)
+    if a just oriented edge leads to a potential articulation point.
+
+    :return:
+    """
+    articulation_vertex = edge.other_end(parent)
+    result = list()
+    for e in articulation_vertex.incidence:
+        if not e.other_end(articulation_vertex) == parent:
+            result += (edge, e)
+    return result
+
+
+def check_edge_direction_event_tree(edge: Edge, origin_vertex: Vertex):
```

Comment:
The name check implies only a read operation. I would call something allong the lines of swap

Reply:
Changed the method name